### PR TITLE
Full Pad portability (Export/Import)

### PR DESCRIPTION
**Pull request by JohnMcLear**
- [x] Requires export of Author data
- [x] Make it work cross server (requires All data, weirdly read only pad id, author data etc. is not enough but not getting an error telling me what's missing) -- help here appreciated :)
- [x] Requires a way to prevent import of "mature" pads
- [ ] Requires testing -- help here appreciated :)
- [x] Requires a way of updating the users UI without a refresh (currently we nuke all Changesets so a user might think they are further into the pad) -- help here appreciated :)

Do not merge, do contribute =)
- [x] Do throw kittens

### CODE REVIEW

File: src/node/handler/ImportHandler.js [hunk 1]
```diff
@@ -116,9 +119,33 @@ exports.doImport = function(req, res, padId)
...
     },
+    function(callback) {
+      var fileEnding = path.extname(srcFile).toLowerCase()
+      var fileIsEtherpad = (fileEnding === ".etherpad");
+
+      if(fileIsEtherpad){
+        // we do this here so we can see if the pad has quit ea few edits
+        padManager.getPad(padId, function(err, _pad){
+          var headCount = _pad.head;
+          if(headCount >= 10){
```
Comment (marcelklehr) [inline src/node/handler/ImportHandler.js]: Why 10?

Comment (JohnMcLear) [inline src/node/handler/ImportHandler.js]: Seemed like a sensible constant.

File: src/node/utils/ExportEtherpad.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        // Get the Pad Authors
+        if(entry.pool && entry.pool.numToAttrib){
+          var authors = entry.pool.numToAttrib;
+          async.forEachSeries(Object.keys(authors), function(k, c){
+            if(authors[k][0] === "author"){
+              var authorId = authors[k][1];
+
+              // Get the author info
+              db.get("globalAuthor:"+authorId, function(e, authorEntry){
+                if(!e) data["globalAuthor:"+authorId] = authorEntry;
```
Comment (marcelklehr) [inline src/node/utils/ExportEtherpad.js]: Possible security hole (reported here: https://groups.google.com/forum/#!topic/etherpad-lite-dev/X31IhThTtIM)

Comment (marcelklehr) [inline src/node/utils/ExportEtherpad.js]: Might need a filter akin to this one: https://github.com/ether/etherpad-lite/blob/develop/src/node/handler/PadMessageHandler.js#L1079

**Comment by JohnMcLear** [comment]
@marcelklehr I'd take that filter if you can pop it in please :)

**Pull request merged**